xls2xml/xls2xml/utils.py

```diff
@@ -19,3 +23,135 @@ def header_to_xml_tag(header):
     # cannot contain spaces (same as above)
 
     return tag
+
+def validate_file(reader, validation_schema_filename):
+    """
+    Validate a data file (XLS or TSV)
+
+    :param reader: reader for the file (XLS or TSV)
+    :type reader: Reader
+    :param validation_schema_filename: path to validation rules definition file
+    :type validation_schema_filename: basestring
+    :return: False if there is error or True for success
+    :rtype: bool
+    """
+    has_validation_error = False
+    validator = MetadataValidator(validation_schema_filename)
+    keys = reader.get_valid_keys()
+    for key in keys:
```

Comment:
If the keys are empty it has a validation error.like it has no valid sheets

Reply:
This method only concerns about validating data rows with a provided validation schema. If the keys are empty that means there is no worksheet to be validated so there is no validation error reported. For the case that there is no valid worksheet, I have added is_valid() method to check all the worksheets contain the required headers. Fixed in 27f748bc